src/core/Akka.Remote.Tests.MultiNode/NewRemoteActorSpec.cs

```diff
@@ -0,0 +1,171 @@
+    {
+        #region Internal actor classes
+
+        public class SomeActor : UntypedActor
+        {
+            protected override void OnReceive(object message)
+            {
+                if(message.Equals("identify"))
+                    Sender.Tell(Self);
+            }
+        }
+
+        public class SomeActorWithParam : UntypedActor
+        {
+            private string _ignored;
+
+            public SomeActorWithParam(string ignored)
+            {
+                _ignored = ignored;
+            }
+
+
+            protected override void OnReceive(object message)
+            {
+                if (message.Equals("identify"))
+                    Sender.Tell(Self);
+            }
+        }
+
+        #endregion
+
+        readonly RoleName _master;
+        public RoleName Master { get { return _master; } }
+        readonly RoleName _slave;
+        public RoleName Slave { get { return _slave; } }
+
+        public NewRemoteActorMultiNodeSpecConfig()
+        {
+            _master = Role("master");
+            _slave = Role("slave");
+
+            CommonConfig =
+                DebugConfig(false)
+                    .WithFallback(ConfigurationFactory.ParseString("akka.remote.log-remote-lifecycle-events = off"));
+
+            DeployOn(Master, @"
+                /service-hello.remote = ""@slave@""
+                /service-hello-null.remote = ""@slave@""
+                /service-hello3.remote = ""@slave@""
+            ");
+
+            DeployOnAll(@"/service-hello2.remote = ""@slave@""");
+        }
+    }
+
+    public class NewRemoteActorSpecNode1 : NewRemoteActorSpec { }
+    public class NewRemoteActorSpecNode2 : NewRemoteActorSpec { }
+
+    public class NewRemoteActorSpec : MultiNodeSpec
+    {
+        private NewRemoteActorMultiNodeSpecConfig _config;
+
+        public NewRemoteActorSpec()
+            : this(new NewRemoteActorMultiNodeSpecConfig())
+        {
+        }
+
+        public NewRemoteActorSpec(NewRemoteActorMultiNodeSpecConfig config) : base(config)
+        {
+            _config = config;
+        }
+
+        protected override int InitialParticipantsValueFactory
+        {
+            get { return Roles.Count; }
+        }
+
+        protected override bool VerifySystemShutdown
+        {
+            get { return true; }
+        }
+
+        [MultiNodeFact]
+        public void NewRemoteActorSpecs()
+        {
+            ANewRemoteActorMustBeLocallyInstantiatedOnARemoteNodeAndBeAbleToCommunicateThroughItsRemoteActorRef();
+            ANewRemoteActorMustBeLocallyInstantiatedOnARemoteNodeWithNullParameterAndBeAbleToCommunicateThroughItsRemoteActorRef();
+            ANewRemoteActorMustBeAbleToShutdownSystemWhenUsingRemoteDeployedActor();
+        }
+        
+        public void ANewRemoteActorMustBeLocallyInstantiatedOnARemoteNodeAndBeAbleToCommunicateThroughItsRemoteActorRef()
+        {
+            RunOn(() =>
```

Comment:
@Aaronontheweb I do not understand how this method runs the given block of code on the specified role as the name `RunOn` suggests. Here is the definition of `RunOn` method

```csharp
public void RunOn(Action thunk, params RoleName[] nodes)
{
            if (nodes.Length == 0) throw new ArgumentException("No node given to run on.");
            if (IsNode(nodes)) thunk();
}
```

All the above method is doing the executing the `Action` delegate. Where is the code that executes it inside a specific role? Or am I missing something very basic here?